### DISCUSSION TITLE
How do I limit dbt job to one concurrent run at the time

### DISCUSSION THREAD

**Discussion by podviaznikov** [dagster-io/dagster#19226] [Q&A] [answered]
I have dbt jobs defined `build_schedule_from_dbt_selection`.

I schedule them every 30 minutes. But once in a while execution will take 40 minutes and I will have two overlapping runs.

1. How do I limit each DBT job to have only one active instance?  I've checked this https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#limiting-concurrency-in-data-pipelines, but still a bit confused.

2. Also, is this possible on dagster to run job continuously instead on a cron schedule. Eg first run finishes in 25 minutes and will kick off next run, which might take 40 minutes and so on. Should this be done with jobs and sensors? Basically finish of the job kicks of another run. Or there some other configs?

**Comment by the4thamigo-uk** [comment]
re: 1 - I also was pretty confused by this. Couple of things... You can limit concurrency at the run level using tags on the job. The docs are here https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#limiting-specific-runs-using-tags. 

```
my_job = define_asset_job(
    name="my_job",
    selection=[my_dbt_asset],
    tags={"my_tag": "my_tag_value"},
)
```

Then in your `dagster.yaml` you need : 

```
run_queue:
  tag_concurrency_limits:
    - key: "my_tag"
      value: "my_tag_value"
      limit: 1
```

(This is fine for local development. In kubernetes deployment you need to set this up in a helm value)

Watch out though, I might be wrong but I think that the concurrency limit doesn't apply to manually launched backflll jobs, but your schedule/sensor launched jobs should be restricted to one at a time. They will be queued.

**Comment by rexledesma** [comment, marked as answer]
For (1), see https://github.com/dagster-io/dagster/discussions/19226#discussioncomment-8150245. 

For (2), see https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors. You could define a run status sensor that checks for your job to finish and then initiates another launch of that job.
